### PR TITLE
Update template

### PR DESCRIPTION
**Pull request by nitosan**
Updated template project for GearVRF
1. Create readme for developer
2. support testing with Google DayDream
3. Fix build errors
4. Separate MainScene code with MainActivity code.
5. add .gitignore file, so when developer made a copy they will have a valid .gitignore file to start with.

### CODE REVIEW

File: template/GVRFApplication/app/src/main/res/values/styles.xml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    <style name="AppBaseTheme" parent="android:Theme.Light">
+        <!--
+            Theme customizations available in newer API levels can go in
+            res/values-vXX/styles.xml, while customizations related to
+            backward-compatibility can go here.
+        -->
+    </style>
+
+    <!-- Application theme. -->
+    <style name="AppTheme" parent="AppBaseTheme">
```
Comment (liaxim) [inline template/GVRFApplication/app/src/main/res/values/styles.xml]: I suggest to use the GVRfAppTheme that the framework defines already. It is a theme that keeps the screen black whenever the app is launched. Some apps can flash a white screen launch time. For example try gvr-immersevipedia with and without the theme.

File: template/GVRFApplication/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -5,6 +5,11 @@
 
     <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
     <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE"/>
+    
+    <!-- Indicates use of Android's VR-mode, available only on Android N+. -->
```
Comment (nitosan) [inline template/GVRFApplication/app/src/main/AndroidManifest.xml]: Got it, will remove these fields

**Comment by nitosan** [comment]
Updated based on feedback

**Comment by liaxim** [comment]
Looks good to me. Planning to merge on Thursday.

**Pull request merged**